docs: rework CLI reference (#4404)
biomejs/website: src/content/docs/reference/cli.mdx

  * [`biome lsp-proxy`](#biome-lsp-proxy)
  * [`biome migrate`](#biome-migrate)
  * [`biome migrate prettier`](#biome-migrate-prettier)
  * [`biome migrate eslint`](#biome-migrate-eslint)
  * [`biome search`](#biome-search)
  * [`biome explain`](#biome-explain)
  * [`biome clean`](#biome-clean)

[//]: # (End-codegen)

## `biome version`

Prints the Biome CLI version and information about the connected daemon server, if any, then exits.

This command also accepts [global flags](#global-flags).

## `biome upgrade`

Upgrades Biome to the latest stable version.

This command upgrades the current Biome binary based on how it was installed:

- **Standalone**: Upgrades a standalone binary in place to the latest stable version.
- **Homebrew**: If Biome was installed with Homebrew, this command runs `brew upgrade biome`.

Set `BIOME_DISTRIBUTION` to `npm`, `homebrew`, or `standalone` to override automatic detection.

:::note
This command cannot update Biome binaries installed through npm. Update `@biomejs/biome` with your package manager instead.
:::

## `biome rage`

Prints a troubleshooting report with CLI, platform, environment, configuration, workspace, and daemon server information.

> This command also accepts [global flags](#global-flags), the relevant [shared flags](#shared-flags), and the following additional flags:

Reviewer: When you say the following *relevant* shared flags, will it be obvious which of the ones in the Shared Flags section are relevant?

I wonder if, when we use this line, we should reiterate using the `help` command, like:

```suggestion
This command also accepts [global flags](#global-flags), the relevant [shared flags](#shared-flags), and the following additional flags. Use the help command (`-h`) to see a complete list.
```

Author: I feel like maybe it would be repeating that a bit too much? Like, for every command basically. No strong feelings either, but I thought it was fine given the nudge to do this in the initial tip block.

#### `--formatter`

Includes the formatter settings loaded from the Biome configuration.

#### `--linter`

Includes the linter settings and enabled lint rules loaded from the Biome configuration.

## `biome start`

Starts the Biome daemon server used for integration with third-party clients, such as editor extensions.

This command also accepts the relevant [shared flags](#shared-flags).

## `biome stop`

Stops the Biome daemon server if it is running.

## `biome check`

Checks the specified files for formatting, linting, and assist actions.

The optional `PATH` arguments accept one or more paths to files or directories. If omitted, Biome processes files in the current working directory.

```shell
biome check ./src
```

This command also accepts [global flags](#global-flags), the relevant [shared flags](#shared-flags), and the following additional flags:

<h3>Flags</h3>

#### `--write`, `--fix`

Applies formatting changes, safe lint fixes, and safe assist actions.

#### `--unsafe`

Allows `--write` or `--fix` to apply unsafe fixes and assist actions.

#### `--formatter-enabled=<true|false>`

Enables or disables formatting checks for this command.

```shell

> biome check --formatter-enabled=true

Reviewer: Just checking, isn't true the default for `check`? Is someone more likely to enable true or enable false when running this command? (goes for all of these)

```

#### `--linter-enabled=<true|false>`

Enables or disables linting checks for this command.

```shell
biome check --linter-enabled=true
```

#### `--assist-enabled=<true|false>`

Enables or disables assist actions for this command.

```shell
biome check --assist-enabled=true
```

#### `--enforce-assist=<true|false>`

Enforces assist actions and causes the command to fail if required actions are not applied. Defaults to `true`.

```shell
biome check --enforce-assist=false
```

#### `--profile-rules`

Enables rule profiling output. Captures timing only for rule execution, not preprocessing such as querying or building the semantic model.

#### `--stdin-file-path=PATH`

Reads code from standard input and writes the processed code to standard output.

> Biome uses `PATH` to select settings and determine the input type from its extension. Virtual paths don't need to exist or belong to the project and bypass `files.includes` and VCS ignore checks.

Reviewer: This might be a little *too* dense? It feels like maybe this second sentence is trying to combine too many ideas, and could be split into two for more comfortable reading.

```shell
echo 'let a;' | biome check --stdin-file-path=file.js --write
```

#### `--staged`

> Processes only staged files. This option is intended for local use.

Reviewer: I wonder whether "processes" is too generic here? This is about `check`ing, so even saying "Checks only staged files" feels more descriptive?

#### `--changed`

> Processes only files changed by commits since the merge base between `REF` and `HEAD`. `REF` comes from `--since` or `vcs.defaultBranch`. Staged and unstaged changes are not included. This option is intended for CI.

Reviewer: This feels a bit technical/jargony, even if extremely accurate!

Maybe closer to... (make this accurate) ?

```suggestion
Processes only files with committed changes compared to your `defaultBranch` or a base reference specified with `--since`. Staged and unstaged changes are not included. This option is intended for CI.
```

This command also accepts the relevant [shared flags](#shared-flags).

## `biome migrate`

Previews configuration updates required by breaking changes. The `prettier` and `eslint` subcommands instead import settings from those tools.

By default, this command displays the proposed changes without modifying files. Use `--write` to apply them.

This command also accepts [global flags](#global-flags), the relevant [shared flags](#shared-flags), and the following additional flags:

<h3>Flags</h3>

#### `--write`, `--fix`

Applies the proposed migration or imported settings.

<h3>Subcommands</h3>

#### `biome migrate prettier`

Imports Prettier configuration and ignore settings into the Biome configuration.

#### `biome migrate eslint`

Imports an ESLint configuration and ignore settings from the current working directory into the Biome configuration.

The `eslint` subcommand accepts these additional flags:

##### Flags

###### `--include-inspired`

Includes Biome rules inspired by ESLint rules in the migration.

###### `--include-nursery`

Includes nursery rules in the migration.

## `biome search`

> Finds code that matches a GritQL pattern. This command is experimental.

Reviewer: It feels like "Experimental" should be the first thing someone sees. 

Maybe keep the original "Experimental"?

```suggestion
EXPERIMENTAL: Finds code that matches a GritQL pattern, optionally limited to a specific file path.
```

Also noting, below you mention `PATH` and `PATTERN` arguments, but I don't think these are named and defined anywhere, nor is there an example of using a path. So, maybe adding that concept here is helpful.

GritQL delimits code snippets with backticks. Because many shells interpret backticks specially, enclose the complete query in single quotes when the shell supports them.

> The required `PATTERN` argument is the GritQL pattern to find. It must be a search pattern because rewrites are not supported.

Reviewer: I wonder whether the whole use single quotes advice could be more simple, something more like this style... (Do this... to avoid...)
```suggestion
Provide the GritQL pattern to find. It must be a search pattern because rewrites are not supported. Put your GritQL query inside single quotes when the shell supports them. This avoids interpreting inline code as commands.
```
NB: I don't know what "it must be a search pattern because rewrites are not supported" means. Is there a risk of people not writing their query correctly? Are people expecting rewrites? (Maybe all that is yes, just saying that I don't know what that means or why/how much it's important, so you need to make the call.)

Also, I'm not sure of the significance of "the complete query". Does this mean, if there's a path, the path needs to be enclosed too? I think it would be assumed that "your query" (i.e. the pattern) means the COMPLETE thing you're searching for, and wouldn't need to be said. It just makes me wonder what you mean by complete. 😄

Author: From my understanding, the GritQL syntax allows you both to search for code or to search and replace (rewrite) them, kind of like regex if you know it. I don't think people would expect rewrites to work here since the command is about *search*. 

But I guess someone could use a pattern that would rewrite the code expecting it to just work or be converted to a search one. I don't know!

> The optional `PATH` arguments limit the search to one or more files or directories. If omitted, Biome searches files in the current working directory.

Reviewer: This sounds great, but a) see my question above re: is this part of a "full" query? and b) can we have an example that uses a path (which would then answer a) if we could see it in the example)?

How about an example that would find all `console.log` invocation in X directory? Then we see what the full command looks like. The default, excluding it and just searching in the current working directory, is trivial by comparison.

```shell
biome search '`console.log($message)`' # find all `console.log` invocations
```

This command also accepts [global flags](#global-flags), the relevant [shared flags](#shared-flags), and the following additional flags:

<h3>Flags</h3>

#### `-l`, `--language=<css|javascript|json>`

Selects the language grammar used for the pattern and searched code: CSS, JavaScript, or JSON.

GritQL patterns are specific to their target grammar, so a search cannot target multiple languages at once.

Defaults to `javascript`.

```shell
biome search --language=javascript '`console.log($message)`'
```

> ## `biome explain`

Reviewer: Nicely done on this one!

Prints documentation for a lint rule or the path to the daemon server log directory.

`NAME` is a lint rule name or `daemon-logs`.

```shell
biome explain noDebugger
biome explain daemon-logs
```

## `biome clean`

Removes the Biome daemon server log files.

## Global flags

> These flags are available to many commands and control general CLI behavior, diagnostics, and output.

Reviewer: ```suggestion
These flags are available to all commands and control general CLI behavior, diagnostics, and output.
```
Is it "all" if they are global?

Author: I think technically some don't apply to `biome version` and `biome help`, but I don't mind being a bit innacurate in that matter if you think that's OK!

| `gitlab` | A [GitLab Code Quality report](https://docs.gitlab.com/ci/testing/code_quality/#implement-a-custom-tool). |
| `junit` | JUnit XML for test and CI integrations. |
| `checkstyle` | [Checkstyle XML](https://checkstyle.org/). |
| `rdjson` | [Reviewdog diagnostic JSON](https://deepwiki.com/reviewdog/reviewdog/3.2-reviewdog-diagnostic-format). |
| `sarif` | [SARIF](https://sarifweb.azurewebsites.net/) for static analysis integrations. |

:::caution
The `json` and `json-pretty` reporters are experimental and may change in patch releases.
:::

#### `--reporter-file=PATH`

Writes the associated reporter's output to `PATH`. Without `--reporter`, it writes the default reporter's output.

```shell
biome ci --reporter=junit --reporter-file=report.xml
```

Repeat `--reporter` to produce multiple outputs. This example prints the default output to the terminal and writes an RDJSON report to a file:

```shell
biome ci --reporter=default --reporter=rdjson --reporter-file=report.json
```

## Shared flags

These flags are accepted by multiple commands. Configuration flags override the equivalent settings from `biome.json` or `biome.jsonc`.

> <h3 id="daemon-flags">Daemon server flags</h3>

Reviewer: There are *so* many shared flags, that I think it's helpful to have these categories visible in the TOC for easier discovery of what exists here. Yes, it will make the TOC longer, but I think it's easier scanning and navigating!

#### `--log-file=PATH`

For file-processing commands, writes internal CLI logs to this file instead of standard output.

Uses the **`BIOME_LOG_FILE`** environment variable.

#### `--log-prefix-name=STRING`

When starting the daemon server, sets the file name prefix used for rotated log files.

Uses the **`BIOME_LOG_PREFIX_NAME`** environment variable.

Defaults to `server.log`.

#### `--log-path=PATH`

When starting the daemon server, sets the directory where log files are stored.

Uses the **`BIOME_LOG_PATH`** environment variable.

Defaults to the `biome-logs` directory inside Biome's platform-specific cache directory.

#### `--log-level=<none|tracing|debug|info|warn|error>`

For file-processing commands, sets the internal CLI logging level to `tracing`, `debug`, `info`, `warn`, `error`, or `none`.

Uses the **`BIOME_LOG_LEVEL`** environment variable.

Defaults to `none`.

#### `--log-kind=<pretty|compact|json>`

For file-processing commands, selects the internal CLI log format: `pretty`, `compact`, or `json`.

## Useful information
Uses the **`BIOME_LOG_KIND`** environment variable.

> - When encountering symbolic links, the CLI will expand them until three levels deep. Deeper levels will result into an error diagnostic.

Author: I've decided to remove this information from here into the configuration file, felt very out of place.

Note that "error diagnostic" doesn't seem to be accurate from what I saw in the core repository; it's meant to be a warning diagnostic instead.
